Footer 5 (#47)

Mikiyas-hs opened the pull request:
## What does this change?

Resolves issue #47 

In this user story, I created the responsive footer section using semantic HTML and a mobile-first CSS approach. The footer includes proper structure, accessible markup, and uses the project’s custom properties for consistent colors.

### Images
<img width="1919" height="595" alt="image" src="https://github.com/user-attachments/assets/0eaa5bdc-6fbb-4836-b8e3-852727d69161" />

<img width="353" height="716" alt="image" src="https://github.com/user-attachments/assets/cc4bed98-ff49-4608-91a8-80d637b5e3af" />


## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] [User test]()
- [x] [Accessibility test]()
- [x] [Performance test]()
- [x] [Responsive Design test]()
- [ ] [Device test]()
- [x] [Browser test]()


## How to review

1. Go to the branch for this PR and open the project in the browser.
2. Compare the footer layout with the design in Figma.
3. Resize the browser window to check the responsive behavior (mobile → tablet → desktop).
4. Verify that:
   - accessible via keyboard.
   - The background color matches the custom properties `--color-neutral`.
   - Text colors use the correct custom properties.
   - The footer remains aligned correctly with other components on different screen sizes.

Mikiyas-hs committed:
feat: add responsive footer with semantic HTML and CSS custom properties (#47) (#56)

Added a new responsive footer using semantic HTML and CSS custom properties. The layout follows a mobile-first approach with nested media queries

Mikiyas-hs committed:
build: added imports & exports (#47)

vsheo commented:
the footer looks good and it also fully responsive

Verify that:
- accessible via keyboard.
  - it is keyboard accessible
- The background color matches the custom properties --color-neutral.
- Text colors use the correct custom properties.
  - youve used the correct custom properties
The footer remains aligned correctly with other components on different screen sizes.
  - it is alligned correctly, but for desktop the there is a bit too much whitespace in the middle of the footer
  - in the browser ive tried to make it look more like the design
 
<img width="2372" height="715" alt="image" src="https://github.com/user-attachments/assets/e297a51d-4228-4f85-9fbf-ab994c392225" />

- the max-width on community-text coul be a bit wider, i tried 50ch, this makes it look better on desktop
- and footer-nav can be changes to flex direction row on wider screens
-  i gave footer-container justify content center
- i gave footer-left, margin right to make more space in the middle
- then i gave footer-nav a bigger gap, and whitespace:nowrap to keep the text on 1 line
This should only be done for the dektop version, mobile en tablet are already good

Mikiyas-hs committed:
style: adjust footer layout and spacing for desktop (#47) (#83)

Updated desktop layout by centering footer content, increasing text width and improving spacing between nav links

vsheo commented:
it look s better now

vsheo committed:
Merge branch 'dev' into footer-5
